lib/pliny/middleware/canonical_log_line.rb

```diff
@@ -0,0 +1,121 @@
+module Pliny::Middleware
+  # Emits a "canonical log line", i.e. a single log line that contains as much
+  # relevant information about a request as possible and which makes for a
+  # single convenient reference point to understand all the vitals of any
+  # single request.
+  #
+  # This default implementation contains some useful data to get a project
+  # started, but it's usually recommended to vendor this middleware into your
+  # project and start adding some custom fields. Some examples of those might
+  # be:
```

Comment:
@brandur I like the idea of the includes in the `LogLine` class. I'm not too keen on how messy the middleware becomes though.

Back to your earlier comment, bringing this in as-is, weed out bugs if present, then vendor it into the template to allow extensibility. I'm starting to think that's probably the best way forward. We could add it to the template by duplicating it, and adding deprecation warnings to the `Pliny::Middleware::CanonicalLogLine` with instructions on how to migrate. How does that sound?